oe_content.post_update.php

```diff
@@ -129,3 +130,47 @@ function oe_content_post_update_00004(): void {
     }
   }
 }
+
+/**
+ * Add new Author sub entity type with required fields.
+ */
+function oe_content_post_update_30005(): void {
+  $entity_type_manager = \Drupal::entityTypeManager();
+  // Install new 'Author' and 'Author type' entity types.
+  $update_manager = \Drupal::entityDefinitionUpdateManager();
+  $update_manager->installEntityType($entity_type_manager->getDefinition('oe_author'));
+  $update_manager->installEntityType($entity_type_manager->getDefinition('oe_author_type'));
+
+  $file_storage = new FileStorage(drupal_get_path('module', 'oe_content') . '/config/post_updates/30005_create_oe_author_entity_type');
+
+  // Create Author bundle.
+  $entity_type_manager->getStorage('oe_author_type')
+    ->create(_oe_content_config_import_prepare($file_storage->read('oe_content.oe_author_type.oe_corporate_body')))
+    ->save();
+
+  // Create field for Author entity type.
+  $field_storage_config = $entity_type_manager->getStorage('field_storage_config');
+  $field_storage_config->create(_oe_content_config_import_prepare($file_storage->read('field.storage.oe_author.oe_skos_reference')))->save();
+  $field_config = $entity_type_manager->getStorage('field_config');
+  $field_config->create(_oe_content_config_import_prepare($file_storage->read('field.field.oe_author.oe_corporate_body.oe_skos_reference')))->save();
+
+  // Configure entity form display.
+  $entity_type_manager->getStorage('entity_form_display')->createFromStorageRecord(_oe_content_config_import_prepare($file_storage->read('core.entity_form_display.oe_author.oe_corporate_body.default')))->save();
+
+  // Create field storage for nodes.
+  $field_storage_config->create(_oe_content_config_import_prepare($file_storage->read('field.storage.node.oe_authors')))->save();
+
+  $content_types = [
+    'oe_event',
+    'oe_news',
+    'oe_page',
+    'oe_policy',
+    'oe_publication',
```

Comment:
We have a problem here: each one of this ctype is provided by an optional submodule, which means that we cannot assume these content type to be available on every site using the oe_content module.

Please move the update hooks in each submodule. We also need to make sure that oe_content update hooks run before the ones of the submodules, not sure how to enforce that.